django/db/backends/mysql/compiler.py

```diff
@@ -36,7 +38,22 @@ def as_sql(self):
 
 
 class SQLUpdateCompiler(compiler.SQLUpdateCompiler, SQLCompiler):
-    pass
+    def as_sql(self):
+        update_query, update_params = super().as_sql()
+        if self.query.order_by:
+            order_by_sql = []
+            order_by_params = []
+            try:
+                for _, (sql, params, is_ref) in self.get_order_by():
+                    order_by_sql.append(sql)
+                    order_by_params.extend(params)
+                update_query += ' ORDER BY ' + ', '.join(order_by_sql)
+                update_params += tuple(order_by_params)
+            except FieldError:
+                # Certain cases can not resolve self.get_order_by().
+                # If this is the case, we will throw an IntegrityError.
+                raise IntegrityError
```

Comment:
You don't need to raise an `IntegrityError`, database will do this for us if necessary.

Reply:
To test if `IntegrityError` has been raised in this [suggestion](https://github.com/django/django/pull/13088#discussion_r448289837), I felt this would be the way to do it.  Otherwise, that particular test fails.  

I can revert that change back to `pass` when handling the `FieldError`, but I'm unsure how to follow through with the suggestion linked above regarding the test.

Reply:
`IntegrityError` will be raised by a database, because `ORDER BY` clause is omitted in this case. We don't need to do this manually.

Reply:
It seems to accepting the query and not throwing the `IntegrityError` without the `ORDER_BY` clause.

https://djangoci.com/job/pull-requests-bionic/database=mysql,label=bionic-pr,python=python3.8/6939/testReport/junit/update.tests/MySQLUpdateOrderByTest/test_order_by_update_on_unique_constraint_with_annotation/

Is there something that's missing or off about `test_order_by_update_on_unique_constraint_with_annotation` that should be causing an `IntegrityError` to be raised?

Reply:
Yes, see https://github.com/django/django/pull/13088/files#r450658069.

Reply:
Ah, thank you for pointing that out! :)